src/pages/StatusPage/index.tsx

```diff
@@ -3,69 +3,86 @@ import { PageStatus, StatusRow } from '../../components';
 import { begin } from '../../services';
 import { Api, ApiWithResponse, Response } from '../../typings/Api';
 import { Container } from './StatusPage.elements';
-
+ 
 interface StatusPageProps {
-  /** An array of API objects */
-  apis: Api[];
-  /** The interval at which to call the APIs in milliseconds */
-  interval?: number;
+ /** An array of API objects */
+ apis: Api[];
+ /** The interval at which to call the APIs in milliseconds */
+ interval?: number;
 }
-
+ 
 interface ApiWithError extends ApiWithResponse {
-  hasError: boolean;
+ hasError: boolean;
 }
-
+ 
 interface StatusPageState {
-  apis: ApiWithError[];
-  hasError: boolean;
+ apis: ApiWithError[];
+ hasError: boolean;
 }
-
+ 
 class StatusPage extends Component<StatusPageProps> {
-  public state: StatusPageState = {
-    apis: [],
-    hasError: false,
-  };
-
-  public handleError = (api: Api, response: Response) => {
-    const apis = [...this.state.apis];
-    const index = apis.findIndex(stateApi => stateApi.api === api);
+ public state: StatusPageState = {
+   apis: [],
+   hasError: false,
+ };
+ 
+ 
+ public handleResponse = (api: Api, response: Response) => {
+   console.log('handleResponse' + response);
+   const apis = [...this.state.apis];
+   const index = apis.findIndex(stateApi => stateApi.api === api);
+   apis[index].response = response;
+  
+   if (response.code == 200) {
+    apis[index].hasError = false;
+    let hasError = false;
+    apis.forEach(api => {
+     if (api.hasError) hasError = true;
+   });
+ 
+   this.setState({
+     apis,
+     hasError: hasError,
+    });
+   } 
+   else {
     apis[index].hasError = true;
-    apis[index].response = response;
-
-    this.setState(
-      {
-        apis,
-        hasError: true
-      }
-    );
-  };
-
-  public componentDidMount = async () => {
-    const apis = this.props.apis.map(api => ({ api, response: null, hasError: false }));
-
+    
     this.setState({
-      apis,
-      hasError: false,
-    });
-
-    await begin(this.props.apis, this.handleError, this.props.interval, undefined);
-  };
-
-  public render() {
-    return (
-      <Container>
-        <PageStatus hasError={this.state.hasError} />
-        {this.state.apis.map((api, i) => (
-          <StatusRow
-            key={api.api.name + api.api.endpoint}
-            name={api.api.name}
-            hasError={api.hasError}
-            className={i === 0 ? 'first' : ''}
-          />
-        ))}
-      </Container>
-    );
-  }
+       apis,
+       hasError: true,
+     });
+   }
+ };
+ 
+ public componentDidMount = async () => {
+   const apis = this.props.apis.map(api => ({ api, response: null, hasError: false }));
+ 
+   this.setState({
+     apis,
+     hasError: false,
+   });
+ 
+   await begin(this.props.apis, this.handleResponse, this.props.interval);
```

Comment:
So, we still need the `handleError` and the `handleResponse` separately. 

`handleError` will get called specifically if there is an error. 
`handleResponse` will get called if there's not an error (the API is not down or if it came back up).

The logic needs to be:

`handleError` - set error state
`handleResponse` - check if previous error state - if there is, iterate through and see if that API is resolved. If no previous error state, do nothing (it's a good response, nothing to update).

Reply:
Check out the last line on the table: https://github.com/katiawheeler/healthy#functions

Reply:
So I did some change. 
According to your impl Im agree that we still need a way to differentiate Response and Error.
I rename Response with Success.
I propose you to have the handler like this : 

1. handleSuccess -> onSuccess
2. handleError -> onError
3. handleDown -> onDown

Reply:
Can we rename `onDown` to `onApiDown` just so it's clear _what_ is down?